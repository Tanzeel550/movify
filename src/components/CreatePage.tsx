import React from 'react';
import { connect, ConnectedProps } from 'react-redux';
import { startCreateMovie } from '../actions/moviesAsyncActions';
import MovieForm from './FormComponent/MovieForm';
import VerifyEmailBtn from './Utils/VerifyEmailBtn';
import VerifiedComp from './Utils/VerifiedComp';

type StateTypes = {
  auth: { user: { emailVerified: boolean } };
  history: { push: (link: string) => {} };
};

const mapStateToProps = ({ auth, history }: StateTypes) => ({
  emailVerified: auth.user.emailVerified,
  history,
});

const connector = connect(mapStateToProps, { startCreateMovie });
type PropsFromRedux = ConnectedProps<typeof connector>;

export const CreatePage = ({
  history,
  startCreateMovie,
  emailVerified,
}: PropsFromRedux) => {
  return (
    <VerifiedComp emailVerified={emailVerified}>
      <MovieForm
        actionType="add"
        handleFormSubmit={async movie => {
          await startCreateMovie({ movie });
          history.push('/');
        }}
      />
    </VerifiedComp>
  );
};

export default connector(CreatePage);
