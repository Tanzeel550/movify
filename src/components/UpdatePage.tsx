import React from 'react';
import MovieForm from './FormComponent/MovieForm';
import { connect, ConnectedProps } from 'react-redux';
import {
  startDeleteMovie,
  startUpdateMovie,
} from '../actions/moviesAsyncActions';
import { Redirect } from 'react-router';
import { FireDBMovieItem } from '../types/APITypes';
import VerifiedComp from './Utils/VerifiedComp';

type StateTypes = {
  movies: FireDBMovieItem[];
  auth: { user: { emailVerified: boolean } };
};

type PropsType = {
  history: {
    push: (link: string) => {};
  };
  match: { params: { id: string } };
  movie: FireDBMovieItem;
};

const mapStateToProps = (state: StateTypes, props: PropsType) => ({
  movie: state.movies.find(movie => movie.id === props.match.params.id),
  emailVerified: state.auth.user.emailVerified,
  history: props.history,
});

const connector = connect(mapStateToProps, {
  startUpdateMovie,
  startDeleteMovie,
});

type Props = ConnectedProps<typeof connector>;

export const UpdatePage = ({
  movie,
  emailVerified,
  history,
  startDeleteMovie,
  startUpdateMovie,
}: Props) => (
  <VerifiedComp emailVerified={emailVerified}>
    {movie ? (
      <MovieForm
        actionType="update"
        movie={movie}
        handleFormSubmit={async data => {
          await startUpdateMovie({ id: movie!.id, movie: data });
          history.push('/');
        }}
        handleMovieDeletion={async () => {
          await startDeleteMovie({ id: movie!.id });
          history.push('/');
        }}
      />
    ) : (
      <h1 className="u-text-colorized">
        No Movie found on this page. Please go back and update some other Movie
        <button
          className="btn u-btn-colorized"
          onClick={() => <Redirect to="/" />}
        >
          Go Home
        </button>
      </h1>
    )}
  </VerifiedComp>
);

export default connector(UpdatePage);
