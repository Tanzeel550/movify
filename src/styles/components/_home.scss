.home__user--info {
  padding: 70px 0;
  background-color: $purple-color-1;
  margin-bottom: 100px;
  color: white;

  p {
    color: white;
  }
}
