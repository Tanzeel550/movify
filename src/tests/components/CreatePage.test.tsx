import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { CreatePage } from '../../components/CreatePage';

let history,
  startCreateMovie,
  wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

beforeEach(() => {
  history = jest.fn();
  startCreateMovie = jest.fn();
  wrapper = shallow(
    <CreatePage
      emailVerified={true}
      history={{ push: history }}
      startCreateMovie={startCreateMovie}
    />
  );
});

test('should render CreatePage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

export {};
