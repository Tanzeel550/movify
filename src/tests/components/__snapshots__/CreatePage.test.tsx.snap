// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CreatePage correctly 1`] = `
<VerifiedComp
  emailVerified={true}
>
  <MovieForm
    actionType="add"
    handleFormSubmit={[Function]}
  />
</VerifiedComp>
`;
